logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/UploadPrivateAssetUseCase.kt

```diff
@@ -0,0 +1,35 @@
+package com.wire.kalium.logic.feature.asset
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.asset.AssetRepository
+import com.wire.kalium.logic.data.asset.AssetType
+import com.wire.kalium.logic.data.asset.UploadedAssetId
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+
+interface UploadPrivateAssetUseCase {
```

Comment:
What's the plan for how to use this use case?

I would have expected use cases like

SendImageUseCase()
SendVideoUseCase()

Reply:
Yep.

If it is private, public, whatever, whatever, it should not concern the apps (who is consuming the UseCase) to decide.

I guess this should not be exposed to consuming apps. It's only a function that the repository should have.



From lower to higher level (please disconsider possible naming mismatches):

### `AssetApi`

Will take some data in the REST API format and upload to the backend. An almost completely dumb interface in which we can interact with the API through Kotlin code.

### `AssetRepository`

Will map business logic data, maybe provide some utilities like writing and reading `AssetDAO` for caching, and and access `AssetApi` when the time comes to upload it.

**`uploadPrivateAsset` could go in here**

### `SendImageUseCase`, `SendVideoUseCase`, `SendAudioUseCase`, etc.

These can take bytearrays (or similar in order to make Platform-agnostic) into them, and a conversation id.
If needed, they will call the `AssetRepository` to insert assets into DB for sending in the future.
Maybe they can wait for sync to be done, or things like this.
This can try to send a message now, or schedule messages to be sent in the future.

Reply:
Thanks for the comments! And yes, that sounds about right. This shouldn't be wrapped as a public consumption use case. So imo there now a few ways of proceeding now in this PR:

1. Refactoring `UploadPrivateAssetUseCase` to something like the proposed use cases above (`SendImageUseCase`, `SendVideoUseCase`, `SendAudioUseCase`, etc.). However these use cases would have to be incomplete for now as there are still some missing pieces required there, like encryption flow, add Asset Message model objects, etc..
2. Removing the `UploadPrivateAssetUseCase` and just leaving the `assetRepository.uploadAndPersistPrivateAsset` method for now. Then later on, once we have the asset encryption methods in place we can try to add the above mentioned use cases.
3. Declaring the `UploadPrivateAssetUseCase` as internal, so that it is not consumable outside of kalium logic module but can be reused later one by a public use case that requires this private asset upload.

After reading your comments I assume this PR should be about doing something similar like point 2, but I just don't want to jump into the refactor wagon without having a clearer picture of it.